### PR TITLE
feat: Add GitHub Organization Custom Role Resource and Data Source

### LINKED ISSUE #1069: [Feature Request] GitHub Custom Repository Role resource
Hi,
I'd like to request the resource for creating custom repository roles.

https://docs.github.com/en/enterprise-cloud@latest/organizations/managing-peoples-access-to-your-organization-with-roles/managing-custom-repository-roles-for-an-organization

Thanks

### PR DESCRIPTION
**Pull request by cailen**
Resolves #1069

----

### Before the change?

*  There was no ability to create custom roles in GitHub Organizations.

### After the change?

* You can create and manage custom roles
* You can refer to them in a data source

### Other information

N/A

----

### Pull request checklist
- [x] Tests for the changes have been added (for bug fixes / features)
  - [x] Data Source:
  ```
  TF_ACC=1 go test -v ./... -run TestAccGithubOrganizationCustomRoleDataSource
    --- PASS: TestAccGithubOrganizationCustomRoleDataSource (7.03s)
    --- PASS: TestAccGithubOrganizationCustomRoleDataSource/queries_a_custom_repo_role (7.03s)
        --- SKIP: TestAccGithubOrganizationCustomRoleDataSource/queries_a_custom_repo_role/with_an_anonymous_account (0.00s)
        --- SKIP: TestAccGithubOrganizationCustomRoleDataSource/queries_a_custom_repo_role/with_an_individual_account (0.00s)
        --- PASS: TestAccGithubOrganizationCustomRoleDataSource/queries_a_custom_repo_role/with_an_organization_account (7.03s)
        PASS
        ok      github.com/integrations/terraform-provider-github/v5/github     7.958s
     ```
  - [x] Resource:
  ```
  TF_ACC=1 go test -v ./... -run TestAccGithubOrganizationCustomRole
     --- PASS: TestAccGithubOrganizationCustomRole (16.99s)
    --- PASS: TestAccGithubOrganizationCustomRole/creates_custom_repo_role_without_error (4.27s)
        --- SKIP: TestAccGithubOrganizationCustomRole/creates_custom_repo_role_without_error/with_an_anonymous_account (0.00s)
        --- SKIP: TestAccGithubOrganizationCustomRole/creates_custom_repo_role_without_error/with_an_individual_account (0.00s)
        --- PASS: TestAccGithubOrganizationCustomRole/creates_custom_repo_role_without_error/with_an_organization_account (4.27s)
    --- PASS: TestAccGithubOrganizationCustomRole/updates_custom_repo_role_without_error (8.30s)
        --- SKIP: TestAccGithubOrganizationCustomRole/updates_custom_repo_role_without_error/with_an_anonymous_account (0.00s)
        --- SKIP: TestAccGithubOrganizationCustomRole/updates_custom_repo_role_without_error/with_an_individual_account (0.00s)
        --- PASS: TestAccGithubOrganizationCustomRole/updates_custom_repo_role_without_error/with_an_organization_account (8.30s)
    --- PASS: TestAccGithubOrganizationCustomRole/imports_custom_repo_role_without_error (4.42s)
        --- SKIP: TestAccGithubOrganizationCustomRole/imports_custom_repo_role_without_error/with_an_anonymous_account (0.00s)
        --- SKIP: TestAccGithubOrganizationCustomRole/imports_custom_repo_role_without_error/with_an_individual_account (0.00s)
        --- PASS: TestAccGithubOrganizationCustomRole/imports_custom_repo_role_without_error/with_an_organization_account (4.42s)
        PASS
        ok      github.com/integrations/terraform-provider-github/v5/github     24.637s
    ```

- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Added the appropriate label for the given change

### Does this introduce a breaking change?

I do not believe so.

Please see our docs on [breaking changes](https://github.com/octokit/.github/blob/master/community/breaking_changes.md) to help!

- [ ] Yes (Please add the `Type: Breaking change` label)
- [x] No

If `Yes`, what's the impact:  

* N/A

### Pull request type

- Feature/model/API additions: `Type: Feature`
- Updates to docs or samples: `Type: Documentation`

----

### CODE REVIEW

File: github/data_source_github_organization_custom_role_test.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+			resource "github_organization_custom_role" "test" {
+				name        = "tf-acc-test-%s"
+				description = "Test role description"
+				base_role   = "read"
+				permissions = [
+					"reopen_issue",
+					"reopen_pull_request",
+				]
+			}
+	`, randomID)
```
[the remark this answers is not in the document]
Comment (cailen) [reply github/data_source_github_organization_custom_role_test.go]: Thanks for catching that. I believe I've fixed it

File: github/resource_github_organization_custom_role.go [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+		Delete: resourceGithubOrganizationCustomRoleDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
```
[the remark this answers is not in the document]
Comment (cailen) [reply github/resource_github_organization_custom_role.go]: You're right. I removed the var and updated the test case

File: website/docs/d/organization_custom_role.html.markdown [hunk 3]
```diff
@@ -0,0 +1,35 @@
+---
```
Comment (kfcampbell) [inline website/docs/d/organization_custom_role.html.markdown:1]: Can you please update the `github.erb` file with links to these awesome new docs?

Comment (cailen) [reply website/docs/d/organization_custom_role.html.markdown:1]: @kfcampbell yep added new commit. thanks

**Review by kfcampbell — APPROVED** [review]
Confirming that the added integration tests are passing for me. Thank you for contributing!

**Comment by kfcampbell** [comment]
Sorry about the delay here; this slipped through the cracks for me somehow. I'll get this merged and released soon!

File: github/resource_github_organization_custom_role.go [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "The base role for the custom repository role.",
+				ValidateFunc: validateValueFunc([]string{"read", "triage", "write", "maintain"}),
+			},
+			"permissions": {
+				Type:        schema.TypeSet,
+				Required:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				MinItems:    1, // At least one permission should be passed.
```
[the remark this answers is not in the document]
Comment (cailen) [reply github/resource_github_organization_custom_role.go:38]: I believe you may be right about this. I was however unable to get it to post to the API at the time, so I was thinking it did require something, but I may have just passed in the wrong blank value.

**Pull request merged**